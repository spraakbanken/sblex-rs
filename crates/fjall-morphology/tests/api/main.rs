use fjall_morphology::FjallMorphology;
use sblex_services::{morphology, Morphology, MorphologyBuilder};
use tempdir::TempDir;

#[test]
fn build_and_load_morphology() -> eyre::Result<()> {
    let tmp_dir = TempDir::new("test.db")?;
    let mut morph = FjallMorphology::new(tmp_dir.path())?;

    morphology::build_from_path(&mut morph, "assets/testing/saldo.lex")?;
    morph.finish()?;

    let result = morph.lookup("dv")?;
    assert!(result.is_none());

    let result = morph.lookup("dv채ljs")?.unwrap();
    let result_json: serde_json::Value = serde_json::from_slice(&result)?;
    insta::assert_json_snapshot!("lookup__dv채ljs", result_json);

    let result = morph.lookup_with_cont("dv")?;
    let result_json: serde_json::Value = serde_json::from_slice(&result)?;
    insta::assert_json_snapshot!("lookup_with_cont__dv", result_json);

    let result = morph.lookup_with_cont("dv채ljs")?;
    let result_json: serde_json::Value = serde_json::from_slice(&result)?;
    insta::assert_json_snapshot!("lookup_with_cont__dv채ljs", result_json);
    Ok(())
}
