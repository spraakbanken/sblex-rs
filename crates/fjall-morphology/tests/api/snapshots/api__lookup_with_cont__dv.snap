---
source: crates/fjall-morphology/tests/api/main.rs
expression: result_json
---
{
  "a": [],
  "c": "a√§"
}
